src/lib.rs

```diff
@@ -325,153 +326,57 @@ impl Dir {
             };
             // 优先填充文件
             if path.is_file() {
-                let source = fs::File::open(&path)?;
-                let file = File {
-                    size: source.metadata()?.len(),
-                    offset: 0,
-                    start_at: 0,
-                    mime: mime_guess::from_path(&path)
-                        .first_or_octet_stream()
-                        .to_string(),
-                    source: Some(fs::File::open(path.clone())?),
-                };
-                *length += file.size;
-                self.files.insert(name, file);
+                let header = FileHeader::from_path(path)?;
+                length += header.size;
+                self.files.insert(name, header);
             } else if path.is_dir() {
                 // 构造子目录
                 let mut dir = Dir {
                     files: HashMap::new(),
                     dirs: HashMap::new(),
                 };
                 // 填充子目录
-                dir.fill_with(root.as_ref(), &path, length)?;
+                length += dir.fill_with(root.as_ref(), &path)?;
                 self.dirs.insert(name, dir);
             }
         }
-        Ok(())
+        Ok(length)
     }
 }
 
-// FRFS: 只读文件系统类型
+/// FRFS header.
 // root: 根目录'/', 用于存储文件所在位置
 // data_size: 在 base 文件中的 start_at 后所跟的数据大小
 // start_at: 在 base 文件中的开始点，无需序列化
+#[derive(Debug, Serialize, Deserialize)]
+struct Header {
```

Comment:
I think the more accurate name is "FRFSHeader", otherwise it is easy to be confused with the file header